base/move_to_front.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <array>
+#include <cstdint>
+#include <limits>
+
+namespace base
+{
+class MoveToFront
+{
+public:
+  static size_t constexpr kNumBytes = static_cast<size_t>(std::numeric_limits<uint8_t>::max()) + 1;
+
+  MoveToFront();
+
+  // Returns index of the byte |b| in the current sequence of bytes, then, moves |b|
```

Comment:
"Moves |b| to the first position" exactly means that "b will be inserted to the first position and removed from the current one".